jobrunner/cli/flags.py

```diff
@@ -0,0 +1,80 @@
+"""
+Ops utility getting and setting db flags
+"""
+import argparse
+import sqlite3
+import sys
+from dataclasses import dataclass
+
+from jobrunner.lib.database import create_table, get_connection, get_flag, set_flag
+
+
+NOVALUE = object()
+
+
+@dataclass
+class Flag:
+    """Do you have a flaaaaaag?"""
+
+    name: str
+    value: str = None
+
+    @classmethod
+    def from_cli(cls, raw):
+        name, equals, value = raw.partition("=")
+        if value == "":
+            if equals == "=":
+                # flag=, set flag to None
+                value = None
+            else:
+                value = NOVALUE
+        return cls(name, value)
+
+
+def main(flags, create):
+    try:
+        do_flags(flags)
+    except sqlite3.OperationalError as e:
+        if "no such table" in str(e):
+            if create:
+                create_table(get_connection(), "flags")
+                do_flags(flags)
+            else:
+                sys.exit(
+                    "Flags table does not exists. Run command again with --create to create it."
+                )
+        else:
+            raise
+
+
+def do_flags(flags):
+    if not flags:
+        all_flags = get_connection().execute("SELECT name FROM flags")
+        flags = [Flag(f[0], NOVALUE) for f in all_flags]
+    for flag in flags:
+        if flag.value is not NOVALUE:
+            set_flag(flag.name, flag.value)
+        print(f"{flag.name}={get_flag(flag.name)}")
```

Comment:
I don't think this is the best named function and I didn't really understand what it was doing until I saw how the API worked. I think explicitly separating the `get` and `set` operations would end up making this code clearer as well as the API safer, and would mean you could do away with the `NOVALUE` sentinel.

Reply:
Done. It's a bit cleaner